src/main/controllers/ChallengeController.kt

```diff
@@ -41,7 +41,19 @@ class ChallengeController: DefaultController<Challenge>(), RestController<Challe
         throw NotImplementedError()
     }
 
-    fun share(user: UserAccount, request: Request): SOAResult<Challenge> {
-        throw NotImplementedError()
+    fun share(user: UserAccount, request: Request): SOAResult<Transaction> {
+        return DaoService.execute {
+            ValidateApiKeyService.execute(user, request.input["secretKey"] as String)
+
+            val challengeId = request.input["challengeId"] as Int
+            val challenge = Challenge.findById(challengeId)!!
```

Comment:
maybe and a try catch around this and throw that new exception

Reply:
maybe we could have a helper method for this -- like u pass a request and it will either return a challenge or throw the error